### PR TITLE
More examples of the usage of variance of inductives

### PR DESCRIPTION
**Pull request by mattam82**
**Kind:** documentation

This just expands on the documentation of variance for universes, explaining when/why a universe is considered covariant, invariant or irrelevant.

### CODE REVIEW

File: doc/sphinx/addendum/universe-polymorphism.rst
Comment (SkySkimmer) [inline doc/sphinx/addendum/universe-polymorphism.rst:326]:
```suggestion
   Monomorphic Universe i.
```

File: doc/sphinx/addendum/universe-polymorphism.rst
Comment (SkySkimmer) [inline doc/sphinx/addendum/universe-polymorphism.rst:370]: It may be worth saying that there's an implicit side condition that the universe has to be big enough to fit `A`

File: doc/sphinx/addendum/universe-polymorphism.rst
Comment (jfehrle) [inline doc/sphinx/addendum/universe-polymorphism.rst:347]:
```suggestion
The universe of :g:`monad` is invariant due to its use on the left side of an arrow in
```

File: doc/sphinx/addendum/universe-polymorphism.rst
Comment (jfehrle) [inline doc/sphinx/addendum/universe-polymorphism.rst:370]: "can be seen" is imprecise, how about "are"?
"as living" is also imprecise, would "exist" or "are present" be accurate?

**Review by jfehrle — COMMENTED** [review]
A few suggestions.

File: doc/sphinx/addendum/universe-polymorphism.rst
Comment (jfehrle) [reply doc/sphinx/addendum/universe-polymorphism.rst:370]: PS Perhaps "any other universe" -> "all other universes" as well?

File: doc/sphinx/addendum/universe-polymorphism.rst
Comment (Zimmi48) [inline doc/sphinx/addendum/universe-polymorphism.rst:86]:
```suggestion
is given by monoids. We first put ourselves in a mode where every declaration
```

**Comment by Zimmi48** [comment]
This PR now needs a rebase.

**Comment by gares** [comment]
ping

**Comment by gares** [comment]
ping

**Comment by gares** [comment]
last call for 8.13.0

**Comment by ppedrot** [comment]
@mattam82 ping.

**Comment by ppedrot** [comment]
@mattam82 this still needs a rebase.

**Comment by mattam82** [comment]
Yep, inadvertent global push

**Comment by silene** [comment]
The first commit seems to come from #13501. What about the second one? Is it really meant to be part of a documentation pull request?

**Comment by ppedrot** [comment]
I've rebased and removed the clearly spurious commits. Can we get an assignee so that it can eventually be merged?

**Comment by SkySkimmer** [comment]
Some comments are still relevant.

@coq/coqbot-maintainers why did this not get a gitlab pipeline?

**Comment by Zimmi48** [comment]
Probably related to the not found issues for documentation artifacts on other PRs (GitLab was giving 5XX error codes). It seems that https://status.gitlab.com/ recognizes that something is wrong.

**Comment by Zimmi48** [comment]
I'm looking into it more closely because apparently the GitLab branch was not updated at all.

**Comment by Zimmi48** [comment]
The log shows that coqbot did push to GitLab but didn't get an answer. I'll try locating the webhook and resending it.

**Comment by Zimmi48** [comment]
It worked. Most likely the explanation was a GitLab-side issue. What we can do on the coqbot-side is improving the logging in such cases.

**Comment by Zimmi48** [comment]
Very likely an occurrence of https://github.com/coq/bot/issues/111.

**Comment by SkySkimmer** [comment]
@mattam82 ping

**Comment by ppedrot** [comment]
Seriously, it is a shame that this important piece of documentation is blocked by minor nitpicks. @mattam82 don't you want to rebase and address them? I'll assign and merge after because that's really a pity.

**Comment by Alizter** [comment]
I wasn't aware of this, I will have a look.

**Comment by Alizter** [comment]
@mattam82 Can you rebase?

**Comment by jfehrle** [comment]
I rebased and fixed the typos.  If someone will advise on wording for the 2 remaining points, I will update those as well.

File: doc/sphinx/addendum/universe-polymorphism.rst
Comment (JasonGross) [reply doc/sphinx/addendum/universe-polymorphism.rst]: > "can be seen" is imprecise, how about "are"?
> "as living" is also imprecise, would "exist" or "are present" be accurate

No.  The "can" is important, and conveys "it is consistent to assume", not "it is actually the case that".  "Exist" and "are present" are less accurate, even if more precise; the more precise but still accurate version is something like "the universe in which the type of eq is an inhabitant".  I don't see why "living" is imprecise, since "inhabiting" is almost a(n English) synonym and is actually technical jargon.

> PS Perhaps "any other universe" -> "all other universes" as well?

No, the point is that you have to pick a single universe, but any choice will do.

Comment (JasonGross) [reply doc/sphinx/addendum/universe-polymorphism.rst]: (I don't see what's wrong with the current wording, hence cannot suggest any better wording.)

Comment (jfehrle) [reply doc/sphinx/addendum/universe-polymorphism.rst]: How about "proofs of equalities may inhabit any other universe."?  Good, meh, fail?

"other" is essential here?

Comment (JasonGross) [reply doc/sphinx/addendum/universe-polymorphism.rst]: Sounds good, and "other" is not essential and dropping it is, I believe, a strict improvement

**Comment by jfehrle** [comment]
Updated.  This is a step forward, so let's merge.  @Zimmi48 would you care to do the honors?

File: doc/sphinx/addendum/universe-polymorphism.rst
Comment (Zimmi48) [inline doc/sphinx/addendum/universe-polymorphism.rst:84]: Unrelated to the present changes, but since I noticed it:

```suggestion
Inductive types can also be declared universe polymorphic on
```

File: doc/sphinx/addendum/universe-polymorphism.rst
Comment (Zimmi48) [inline doc/sphinx/addendum/universe-polymorphism.rst:437]: There's an indentation problem that makes this example "escape" the example box.

**Review by Zimmi48 — COMMENTED** [review]
Minor things. I notice that @SkySkimmer is assigned though, so maybe he wants to review before merging.

File: doc/sphinx/addendum/universe-polymorphism.rst
Comment (SkySkimmer) [reply doc/sphinx/addendum/universe-polymorphism.rst:84]: I have this in https://github.com/coq/coq/pull/15981/files

**Comment by jfehrle** [comment]
Updated.  If no more comments, please merge.

**Review by Zimmi48 — APPROVED** [review]
@coqbot merge now

**Comment by Zimmi48** [comment]
@coqbot merge now

**Pull request merged**
